src/com/epam/javatrainings/validation/ValidatorUtils.java

```diff
@@ -0,0 +1,32 @@
+package com.epam.javatrainings.validation;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+
+public class ValidatorUtils {
+
+    public static AnnotationProcessor detectAnnotationProcessor(Annotation annotation) {
```

Comment:
The method name says that it detects the expected AnnotationProcessor, but actually, it creates a new AnnotationProcessor on each method call. Can be used this approach for detecting:
```
 public static <T extends Annotation>  AnnotationProcessor detectAnnotationProcessor(Class<T> annotation) {
        AnnotationProcessor annotationProcessor = annotationProcessorsMap.get(annotation);
        if (annotationProcessor == null) throw new IllegalArgumentException("unknown annotation type");
        return annotationProcessor;
    }
```

Reply:
Fixed